### ISSUE TITLE
The changed behavior of StreamingPullFuture.cancel() should be documented

### ISSUE THREAD

**Issue by plamut** [googleapis/python-pubsub#419]
Calling `StreamingPullFuture.cancel()` invokes `close()` on the `StreamingPullManager` instance, but the latter method is now [non-blocking](https://github.com/googleapis/python-pubsub/pull/320/files#diff-2e54d7a3f689e6c74875b7a188385583ef1c74f9baf20362e55ff77f337fde16R534-R535) as of version `v2.4.1`, meaning that `StreamingPullFuture.cancel()` has become non-blocking as well.

This can be problematic if subscriber client is used as a context manager, and cancelling the future is the last action in it. Exiting the `with` block closes the underlying transport, but the streaming pull shutdown might still be in progress. If a, say, NACK request is made during the shutdown after the transport has already been closed, an error occurs ([user comment](https://github.com/googleapis/python-pubsub/issues/176#issuecomment-850357713)).

TL; DR - user should know that `cancel()` is non-blocking, and that the way to block on the shutdown is by calling the `streaming_pull_future.result()` method instead.

Also, at least one of the samples needs to be updated as well, but that's a separate issue.
